invokeai/app/services/image_storage.py

```diff
@@ -71,19 +79,75 @@ def __init__(self, output_folder: str):
                 parents=True, exist_ok=True
             )
 
+    def list(
+        self, image_type: ImageType, page: int = 0, per_page: int = 10
+    ) -> PaginatedResults[ImageResponse]:
+        dir_path = os.path.join(self.__output_folder, image_type)
+        image_paths = glob(f"{dir_path}/*.png")
+        count = len(image_paths)
+
+        # TODO: do all platforms support `getmtime`? seem to recall some do not...
+        sorted_image_paths = sorted(
+            glob(f"{dir_path}/*.png"), key=os.path.getmtime, reverse=True
+        )
+
+        page_of_image_paths = sorted_image_paths[
+            page * per_page : (page + 1) * per_page
+        ]
+
+        page_of_images: List[ImageResponse] = []
+
+        for path in page_of_image_paths:
+            filename = os.path.basename(path)
+            img = PILImage.open(path)
```

Comment:
Is there not a way to get this metadata without actually opening the image? It could be fairly expensive to decode the entire image every time.

Reply:
I don't think so. On the UI you have to request the image (like, over the network) and load it to get it. That feels far less efficient than doing it here. AFAIK there's no FS level way to get width and height. 

When we have an index for images we'd just store it in there.

Reply:
After reviewing PIL documentation, `Image.open()` does not actually load the image into memory until you try to do something: https://pillow.readthedocs.io/en/stable/reference/Image.html#PIL.Image.open